livekit-agents/livekit/agents/stt/fallback_adapter.py

```
@@ -54,6 +54,9 @@ def __init__(
             capabilities=STTCapabilities(
                 streaming=all(t.capabilities.streaming for t in stt),
                 interim_results=all(t.capabilities.interim_results for t in stt),
+                single_frame_recognition=all(
```

I think we should handle this a bit differently.

IMO streaming and recognize should both be supported by fallback adapter. and if one of the plugins don't support a capability, we can use our workaround path.
* if streaming isn't supported, then we should throw an error letting them know they should wrap it with a fallback adapter
* recognize should be supported, you already have a workaround to use streaming to handle recognition

---

I think only one approach is needed, and we shouldn't introduce runtime errors for fallback adapters since users likely have no way to test them.

- if we want to give prefrence to streaming, we can pre-check if any plugin supports it. If so, we can notify users to wrap non-streaming plugins with a streaming adapter.
- If none of the plugins support streaming, we can use recognize method.

In that case, we don't need workaround for recognize method.

---

within agents, we *only* use the streaming method for STT. so this would cause problems if you are creating a fallback adapter that is reporting that it doesn't streaming.. we'd have to wrap everything with a StreamAdapter.. this is not a good idea.

the reason why recognize is used is to recover from a streaming failure.. I don't see why we shouldn't also declare it support it since you are handling it internally.